invenio_communities/assets/semantic-ui/js/invenio_communities/requests/index.js

```diff
@@ -6,46 +6,29 @@
  * under the terms of the MIT License; see LICENSE file for more details.
  */
 
-import React, { Component } from "react";
-import ReactDOM from "react-dom";
-import _camelCase from "lodash/camelCase";
-import { overrideStore } from "react-overridable";
-import { SearchApp } from "@js/invenio_search_ui/components";
-import { defaultComponents as RequestsDefaultComponents } from "./requests";
-
-const domContainer = document.getElementById("app");
-const community = JSON.parse(domContainer.dataset.community);
-
-const getConfigFromDataAttribute = (element, attr) => {
-  const dataValue = domContainer.dataset[attr];
-
-  return JSON.parse(dataValue);
+import React from "react";
+import { createSearchAppInit } from "@js/invenio_search_ui";
+import {
+  BucketAggregationElement,
+  RecordFacetsValues, RecordSearchBarElement, RequestsEmptyResultsWithState,
+  RequestsFacets, RequestsResults,
+  RequestsResultsGridItemTemplate,
+  RequestsResultsItemTemplate,
+  RequestsSearchLayout,
+} from './requests';
+
+
+const defaultComponents = {
+  "BucketAggregation.element": BucketAggregationElement,
+  "BucketAggregationValues.element": RecordFacetsValues,
+  "SearchApp.facets": RequestsFacets,
+  "ResultsList.item": RequestsResultsItemTemplate,
+  "ResultsGrid.item": RequestsResultsGridItemTemplate,
+  "SearchApp.layout": RequestsSearchLayout,
+  "SearchApp.results": RequestsResults,
+  "SearchBar.element": RecordSearchBarElement,
+  "EmptyResults.element": RequestsEmptyResultsWithState,
 };
 
-class CommunityRequests extends Component {
-  constructor(props) {
-    super(props);
-
-    const { appId, ...config } = getConfigFromDataAttribute(
-      domContainer,
-      _camelCase("invenio-config")
-    );
-    this.appId = appId;
-    this.config = config;
-
-    for (const [componentId, component] of Object.entries(
-      RequestsDefaultComponents(appId)
-    )) {
-      overrideStore.add(componentId, component);
-    }
-  }
-
-  render() {
-    return (
-      <SearchApp appName={this.appId} key={this.appId} config={this.config} />
-    );
-  }
-}
-
-ReactDOM.render(<CommunityRequests community={community} />, domContainer);
-export default CommunityRequests;
+// Auto-initialize search app
+createSearchAppInit(defaultComponents);
```

Comment:
Good refactor :+1:, I was wondering why we were not using `createSearchAppInit` for the requests but I was not brave enough to change it  ... thanks!